lagtraj/forcings/conversion/targets/dephy.py

```diff
@@ -480,42 +507,79 @@ def nudging_inv_time_prof(nudging_parameters, variable):
             dephy_variables[variable],
         )
 
-    nudging_parameters_momentum_traj = parameters.nudging_parameters_momentum_traj
-    nudging_parameters_scalar_traj = parameters.nudging_parameters_scalar_traj
-    ds_dephy["nudging_inv_u_traj"] = nudging_inv_time_prof(
-        nudging_parameters_momentum_traj, "nudging_inv_u_traj"
-    )
-    ds_dephy["nudging_inv_v_traj"] = nudging_inv_time_prof(
-        nudging_parameters_momentum_traj, "nudging_inv_v_traj"
-    )
-    ds_dephy["nudging_inv_temp_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_temp_traj"
-    )
-    ds_dephy["nudging_inv_theta_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_theta_traj"
-    )
-    ds_dephy["nudging_inv_thetal_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_thetal_traj"
-    )
-    ds_dephy["nudging_inv_qv_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_qv_traj"
-    )
-    ds_dephy["nudging_inv_qt_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_qt_traj"
-    )
-    ds_dephy["nudging_inv_rv_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_rv_traj"
-    )
-    ds_dephy["nudging_inv_rt_traj"] = nudging_inv_time_prof(
-        nudging_parameters_scalar_traj, "nudging_inv_rt_traj"
-    )
-
+    # First, set the flags
+    if parameters.nudging_method_momentum in [False, None]:
+        nudging_u = 0
+        nudging_v = 0
+    elif parameters.nudging_method_momentum in ["constant", "fixed_height"]:
+        nudging_u = -1
+        nudging_v = -1
+    elif parameters.nudging_method_momentum == "runtime_inversion_height":
+        nudging_u = -2
+        nudging_v = -2
+    else:
+        raise ValueError("nudging_method_momentum value invalid")
+    if parameters.nudging_method_scalars in [False, None]:
+        nudging_temp = 0
+        nudging_theta = 0
+        nudging_thetal = 0
+        nudging_qv = 0
+        nudging_qt = 0
+        nudging_rv = 0
+        nudging_rt = 0
+    elif parameters.nudging_method_scalars in ["constant", "fixed_height"]:
+        nudging_temp = -1
+        nudging_theta = -1
+        nudging_thetal = -1
+        nudging_qv = -1
+        nudging_qt = -1
+        nudging_rv = -1
+        nudging_rt = -1
+    elif parameters.nudging_method_scalars == "runtime_inversion_height":
+        nudging_temp = -2
+        nudging_theta = -2
+        nudging_thetal = -2
+        nudging_qv = -2
+        nudging_qt = -2
+        nudging_rv = -2
+        nudging_rt = -2
+    else:
+        raise ValueError("nudging_method_scalars value invalid")
+    # Make the profiles if needed
+    if parameters.nudging_method_momentum in ["constant", "fixed_height"]:
+        ds_dephy["nudging_inv_u_traj"] = nudging_inv_time_prof(
+            "momentum", "nudging_inv_u_traj"
```

Comment:
I think it would be clearer to have `nudging_inv_u_traj`, `nudging_inv_v_traj` etc as a list here and then a `for`-loop calling `nudging_inv_time_prof()`